src/Workspaces/CSharp/Portable/CodeGeneration/OperatorGenerator.cs

```diff
@@ -86,23 +87,36 @@ internal static class OperatorGenerator
 
             var operatorDecl = SyntaxFactory.OperatorDeclaration(
                 attributeLists: AttributeGenerator.GenerateAttributeLists(method.GetAttributes(), options),
-                modifiers: GenerateModifiers(),
+                modifiers: GenerateModifiers(method),
                 returnType: method.ReturnType.GenerateTypeSyntax(),
+                explicitInterfaceSpecifier: GenerateExplicitInterfaceSpecifier(method.ExplicitInterfaceImplementations),
                 operatorKeyword: SyntaxFactory.Token(SyntaxKind.OperatorKeyword),
                 operatorToken: operatorToken,
                 parameterList: ParameterGenerator.GenerateParameterList(method.Parameters, isExplicit: false, options: options),
                 body: hasNoBody ? null : StatementGenerator.GenerateBlock(method),
+                expressionBody: null,
                 semicolonToken: hasNoBody ? SyntaxFactory.Token(SyntaxKind.SemicolonToken) : new SyntaxToken());
 
             operatorDecl = UseExpressionBodyIfDesired(options, operatorDecl, parseOptions);
             return operatorDecl;
         }
 
-        private static SyntaxTokenList GenerateModifiers()
+        private static SyntaxTokenList GenerateModifiers(IMethodSymbol method)
         {
-            return SyntaxFactory.TokenList(
-                SyntaxFactory.Token(SyntaxKind.PublicKeyword),
-                SyntaxFactory.Token(SyntaxKind.StaticKeyword));
+            var tokens = ArrayBuilder<SyntaxToken>.GetInstance();
+
+            if (method.ExplicitInterfaceImplementations.Length == 0)
+            {
+                tokens.Add(SyntaxFactory.Token(SyntaxKind.PublicKeyword));
+            }
+
+            if (method.IsAbstract)
+            {
+                tokens.Add(SyntaxFactory.Token(SyntaxKind.AbstractKeyword));
+            }
+
+            tokens.Add(SyntaxFactory.Token(SyntaxKind.StaticKeyword));
```

Comment:
i would expect static to come before abstract.  But i guess we have no precedence.  How are these ordered in our default modifiers list?

Reply:
`static` comes before `abstract`

Reply:
Note to self: MethodGenerator may need to adjust the order too:

https://github.com/dotnet/roslyn/blob/781dab82c6cc8a6e0d04e66bcbacf04ab16e2e91/src/Workspaces/CSharp/Portable/CodeGeneration/MethodGenerator.cs#L240-L253